python/sglang/bench_offline_throughput.py

```diff
@@ -0,0 +1,288 @@
+    }
+
+    prompt = [r[0] for r in reqs]
+    sampling_params = {
+        "temperature": 0,
+        "max_new_tokens": output_len,
+        "ignore_eos": ignore_eos,
+    }
+
+    st = time.perf_counter()
+    gen_out = backend.generate(prompt=prompt, sampling_params=sampling_params)
+    latency = time.perf_counter() - st
+
+    if backend_name == "runtime":
+        gen_out = json.loads(gen_out)
+
+    measurement_results["total_latency"] = latency
+    measurement_results["total_output_tokens"] = sum(
+        o["meta_info"]["completion_tokens"] for o in gen_out
+    )
+    measurement_results["throughput"] = (
+        measurement_results["total_input_tokens"]
+        + measurement_results["total_output_tokens"]
+    ) / latency
+
+    print(f"Throughput: {measurement_results['throughput']} tokens/s")
+    return measurement_results
+
+
+def throughput_test(
+    server_args: ServerArgs,
+    bench_args: BenchArgs,
+):
+    if bench_args.backend == "engine":
+        backend = getEngine(**dataclasses.asdict(server_args))
+        if not backend:
+            raise ValueError("Please provide valid engine arguments")
+    elif bench_args.backend == "runtime":
+        backend = Runtime(**dataclasses.asdict(server_args))
+    else:
+        raise ValueError('Please set backend to either "engine" or "runtime"')
+
+    tokenizer_id = server_args.model_path
+    tokenizer = get_tokenizer(tokenizer_id)
+
+    # Set global environmnets
+    set_ulimit()
+    random.seed(bench_args.seed)
+    np.random.seed(bench_args.seed)
+
+    if bench_args.dataset_name == "sharegpt":
+        input_requests = sample_sharegpt_requests(
+            dataset_path=bench_args.dataset_path,
+            num_requests=bench_args.num_prompts,
+            tokenizer=tokenizer,
+            fixed_output_len=bench_args.sharegpt_output_len,
+        )
+        output_len = bench_args.sharegpt_output_len
+    elif bench_args.dataset_name == "random":
+        input_requests = sample_random_requests(
+            input_len=bench_args.random_input_len,
+            output_len=bench_args.random_output_len,
+            num_prompts=bench_args.num_prompts,
+            range_ratio=bench_args.random_range_ratio,
+            tokenizer=tokenizer,
+            dataset_path=bench_args.dataset_path,
+        )
+        output_len = bench_args.random_output_len
+    elif bench_args.dataset_name == "generated-shared-prefix":
+        input_requests = sample_generated_shared_prefix_requests(
+            num_groups=bench_args.gen_num_groups,
+            prompts_per_group=bench_args.gen_prompts_per_group,
+            system_prompt_len=bench_args.gen_system_prompt_len,
+            question_len=bench_args.gen_question_len,
+            output_len=bench_args.gen_output_len,
+            tokenizer=tokenizer,
+        )
+        output_len = bench_args.gen_output_len
+    else:
+        raise ValueError(f"Unknown dataset: {bench_args.dataset_name}")
```

Comment:
This part seems redundant. Can you wrap this common part as a function in `bench_serving.py` and import it her?